client/src/main/java/io/pravega/client/stream/EventWriterConfig.java

```diff
@@ -23,12 +23,22 @@
     private final int maxBackoffMillis;
     private final int retryAttempts;
     private final int backoffMultiple;
-
+    /**
+     * The maximum amount of time, in milliseconds, which a transaction can run before it is considered failed.
+     */
+    private final long transactionTimeoutTime;
+    /**
+     * The maximum amount of time, in milliseconds after a scale operation has been initiated before a transaction is timedout. 
+     */
+    private final long transactionTimeoutScaleGracePeriod;
+    
     public static final class EventWriterConfigBuilder {
         private int initalBackoffMillis = 1;
         private int maxBackoffMillis = 20000;
         private int retryAttempts = 10;
         private int backoffMultiple = 10;
+        private long transactionTimeoutTime = 60 * 60 * 1000;
+        private long transactionTimeoutScaleGracePeriod = 0;
```

Comment:
Zero isn't a legal value, check this out:

https://github.com/pravega/pravega/blob/master/controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java#L355